### ISSUE TITLE
copy() not working with indefinite encodings

### ISSUE THREAD

**Issue by joernheissler** [wbond/asn1crypto#150]
```python
from asn1crypto.core import OctetString
ios = OctetString.load(b'\x24\x80\x04\x03foo\x00\x00')
cpy = ios.copy()
print(cpy.dump())
```

Output: `b'\x04\x05\x04\x03foo'`
Expected: `b'\x24\x80\x04\x03foo\x00\x00'` or `b'\x04\x03foo'`

For the missing `\x00\x00` see #136.
The result is quite broken.
I tried adding `self._header = other._header` to `Asn1Value._copy`, but then `Certificate.authority_issuer_serial` will return a different values causing a test case to fail. I have no idea what `authority_issuer_serial` is for, so I cannot check if perhaps the test case is wrong.

Perhaps `dump` and `_dump_header` should be changed to output indefinite encodings if so desired? But I guess this violates one basic principle of the library: to generate only DER.

**Comment by wbond** [comment]
Yeah, the library should only generate DER encoded-values. However, if we haven't modified the value, it is okay to re-output BER.

Most likely we'll need to tweak the `Constructable._copy()` method to copy the `_header` and `_trailer` if `_indefinite is True`. Another possible approach is to round-trip `.native` into `.set()` to ensure the state is cleared out properly, but that should only be done if `_indefinite is True`.

**Comment by wbond** [comment]
This should be fixed 60136d7dce4b8ae89d5d000d249587806f3d5371
